src/SubscriptionItem.php

```diff
@@ -207,16 +208,37 @@ public function swapAndInvoice($price, array $options = [])
      *
      * @param  int  $quantity
      * @param  \DateTimeInterface|int|null  $timestamp
-     * @return \Stripe\UsageRecord
+     * @return \Stripe\V2\Billing\MeterEvent
      */
     public function reportUsage($quantity = 1, $timestamp = null)
     {
-        $timestamp = $timestamp instanceof DateTimeInterface ? $timestamp->getTimestamp() : $timestamp;
+        // Get the price to determine the meter
+        $stripePrice = $this->subscription->owner->stripe()->prices->retrieve($this->stripe_price);
 
-        return $this->subscription->owner->stripe()->subscriptionItems->createUsageRecord($this->stripe_id, [
-            'quantity' => $quantity,
-            'action' => $timestamp ? 'set' : 'increment',
-            'timestamp' => $timestamp ?? time(),
+        if (! isset($stripePrice->recurring->meter)) {
+            throw new \InvalidArgumentException('Price must have a meter to report usage. Legacy usage records are no longer supported.');
+        }
+
+        // Get the meter to get the event name
+        $meter = $this->subscription->owner->stripe()->billing->meters->retrieve($stripePrice->recurring->meter);
```

Comment:
Just wondering if the event name should be stored locally?

Otherwise for an application that is reporting a lot of usage, the Stripe API might be hit a lot more than is necessary?

Reply:
Great idea, I have added that suggestion.